keps/sig-scalability/0001-enhance-network-performance-benchmark-testcases/README.md

```diff
@@ -0,0 +1,887 @@
+	* 5 Client PODS will send traffic to 1 ClusterIP/NodePort svc IP.
+	* Below testcases will measure the network performance metrics with scaled service IPs (1 ClientPOD -> 1 ServiceIP -> 1 ServerPOD), as well as with load-balancing (N ClientPOD -> 1 ServiceIP -> N ServerPOD).
+	
+     TestcaseId  | Testcase Description | Tool | Metrics Measured
+     ----------- | ---------------------|------|-----------------
+     TC2.2.1.1  | Send HTTP traffic from 1 client pod IPs to ClusterIP. 1 server pod is located behind ClusterIP service (1:1) | siege | Latency , ResponseTime
+     TC2.2.1.2  | Send HTTP traffic from 1 client pod IPs to NodePort svc IP. 1 server pod is located behind NodePort service (1:1) | siege | Latency , ResponseTime
+     TC2.2.1.3  | Send HTTP traffic from 5 client pod IPs to ClusterIP. 5 server pods are located behind ClusterIP service. | siege | Latency , ResponseTime
+     TC2.2.1.4  | Send HTTP traffic from 5 client pod IPs to NodePort svc IP. 5 server pods are located behind NodePort service. | siege | Latency , ResponseTime
+	
+      2.2.2 Client-Server POD on different worker node (N:M) (TCP/UDP traffic)
+	  
+	* Client POD size N=50 and Server POD size M=50. Each server pod will be behind one unique ClusterIP/NodePort Svc. So 50 clusterIP will be created for TCP/UDP traffic only. ( 1 Client POD ---> 1 ClusterIP ---> 1 Server POD)
+	* This test will require 100 worker nodes i.e N+M, so that we can make sure client and server pods are not co-located.
+	* Each metric will be a percentile of 50 pairs.
+		
+     TestcaseId  | Testcase Description | Tool | Metrics Measured
+     ----------- | ---------------------|------|-----------------
+     TC2.2.2.1  | Send TCP traffic from 1 client pod IPs to 1 server pod is located behind respective unique ClusterIP (1:1) | iperf3 | Throughput in Gbits/s
+     TC2.2.2.2  | Send UDP traffic from 1 client pod IPs to 1 server pods is located behind respective unique ClusterIP (1:1) | iperf2 | Packets per second (pps), Loss Rate, Latency, jitter
+     TC2.2.2.3  | Send TCP traffic from 1 client pod IPs to NodePort svc IP. 1 server pod is located behind NodePort service (1:1) | iperf3 | Throughput in Gbits/s
+     TC2.2.2.4  | Send UDP traffic from 1 client pod IPs to NodePort svc IP. 1 server pod is located behind NodePort service (1:1) | iperf2 | Packets per second (pps), Loss Rate, Latency, jitter
+    
+
+### Tools
+
+We intend to use the following tools for testing :
+
+1. iperf3: For TCP throughput measurement.
+
+2. iperf2: For UDP latency, loss rate , pps measurement.
+
+3. Siege: Siege is an open source benchmark utility. 
+   It will be used for measuring HTTP/HTTPs metrics such as total number of hits recorded, 
+   bytes transferred, response time, concurrency, and return status. 
+
+Other tools such as Trex will be used in future for further enhancements.
+
+### Resource Requirements
```

Comment:
@wojtek-t @mm4tt  , have one query. Is VM(worker node) over subscription , a possibility on  compute servers in google-cloud?